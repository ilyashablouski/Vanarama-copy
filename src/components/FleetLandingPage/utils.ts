import Heading from "@vanarama/uibook/lib/components/atoms/heading";

export const prepareTagName = (possibleTag: string | null) =>
    possibleTag && Heading.defaultProps?.tag?.indexOf(possibleTag) !== -1
        ? possibleTag
        : undefined;