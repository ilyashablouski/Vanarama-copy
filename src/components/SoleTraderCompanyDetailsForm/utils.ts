// eslint-disable-next-line import/prefer-default-export
export const calculateDisposableIncome = ({
  annualTurnover,
  existingVehicle,
  annualCostOfSales,
  annualExpenses,
  monthlyAmountBeingReplaced,
}: {
  annualTurnover: string;
  existingVehicle: boolean;
  annualCostOfSales: string;
  annualExpenses: string;
  monthlyAmountBeingReplaced: string;
}) => {
  const totalExpenses = +annualCostOfSales + +annualExpenses;

  if (!existingVehicle) {
    return +annualTurnover - totalExpenses;
  }

  const totalTurnover = +annualTurnover + +monthlyAmountBeingReplaced * 12;
  return totalTurnover - totalExpenses;
};
