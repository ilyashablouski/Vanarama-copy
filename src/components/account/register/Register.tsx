import React, { ChangeEvent } from 'react';
import Input from '@vanarama/uibook/packages/ui-components/src/components/atoms/textinput';
import Button from '@vanarama/uibook/packages/ui-components/src/components/atoms/button';
import FormGroup from '@vanarama/uibook/packages/ui-components/src/components/molecules/formgroup';
import { IRegisterProps, IRegisterState } from './interfaces';

class Register extends React.Component<IRegisterProps, IRegisterState> {
  constructor(props: any) {
    super(props);

    this.state = {
      email: '',
      password: '',
      repeatPassword: '',
    };
  }

  handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const { register } = this.props;
    const { email, password } = this.state;

    register(email, password);
  };

  handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
    e.preventDefault();
    const { name, value } = e.currentTarget;

    if (Object.keys(this.state).includes(name)) {
      this.setState(prevState => ({ ...prevState, [name]: value }));
    }
  };

  render() {
    const { email, password, repeatPassword } = this.state;

    return (
      <section>
        <form onSubmit={this.handleSubmit} id="loginForm" className="form">
          <FormGroup>
            <Input
              id="registerEmail"
              label="Your Email"
              type="text"
              name="email"
              value={email}
              onChange={this.handleInputChange}
            />
          </FormGroup>
          <FormGroup>
            <Input
              id="registerPassword"
              label="Your Password"
              type="password"
              name="password"
              value={password}
              onChange={this.handleInputChange}
            />
          </FormGroup>
          <FormGroup>
            <Input
              id="registerRepeatPassword"
              label="Repeat Password"
              type="password"
              name="repeatPassword"
              value={repeatPassword}
              onChange={this.handleInputChange}
            />
          </FormGroup>
          <FormGroup>
            <Button
              id="registerButton"
              type="submit"
              label="Register"
              color="primary"
            />
          </FormGroup>
        </form>
      </section>
    );
  }
}

export default Register;
