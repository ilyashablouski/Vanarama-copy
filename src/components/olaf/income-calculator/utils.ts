import { IIncomeCalculatorFormValues } from './interfaces';

// eslint-disable-next-line import/prefer-default-export
export const calculateIncome = (values: IIncomeCalculatorFormValues) => {
  const monthlyExpenses =
    Number(values.mortgageOrRent) +
    Number(values.phoneAndInternet) +
    Number(values.creditCardPayments) +
    Number(values.insurance) +
    Number(values.foodAndClothes) +
    Number(values.studentLoans) +
    Number(values.utilities) +
    Number(values.carFinance) +
    Number(values.fuel) +
    Number(values.otherCredit);

  const disposableIncome =
    Number(values.averageMonthlyIncome) - monthlyExpenses;

  return {
    monthlyExpenses,
    disposableIncome,
  };
};
