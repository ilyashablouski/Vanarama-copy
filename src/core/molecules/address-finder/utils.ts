import { ILoqateSuggestion } from '../../../hooks/useLoqate/interfaces';

// eslint-disable-next-line import/prefer-default-export
export const suggestionToDisplay = (suggestion: ILoqateSuggestion) =>
  `${suggestion.text} - ${suggestion.description}`;
