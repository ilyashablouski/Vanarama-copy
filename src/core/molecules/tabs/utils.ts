export const tabId = (index: number, id?: string) => `tab:${id}-${index}`;
export const panelId = (index: number, id?: string) => `panel:${id}-${index}`;
