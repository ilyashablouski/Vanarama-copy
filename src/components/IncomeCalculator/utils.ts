import { IIncomeCalculatorFormValues } from './interfaces';
import { IncomeCalculatorExpenditure } from '../../../generated/IncomeCalculatorExpenditure';
import { sum } from '../../utils/array';

export const calculateMonthlyExpenses = (
  values?: IIncomeCalculatorFormValues | IncomeCalculatorExpenditure | null,
) =>
  sum(
    [
      values?.mortgageOrRent,
      values?.phoneAndInternet,
      values?.creditCardPayments,
      values?.insurance,
      values?.foodAndClothes,
      values?.studentLoan,
      values?.utilities,
      values?.carFinance,
      values?.fuel,
      values?.otherCredit,
    ],
    item => Number(item || ''),
  );

// eslint-disable-next-line import/prefer-default-export
export const calculateIncome = (values: IIncomeCalculatorFormValues) => {
  const monthlyExpenses = calculateMonthlyExpenses(values);

  const disposableIncome =
    Number(values.averageMonthlyIncome) - monthlyExpenses;

  return {
    monthlyExpenses,
    disposableIncome,
  };
};
