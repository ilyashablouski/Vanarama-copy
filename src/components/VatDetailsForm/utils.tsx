import { useCallback, useEffect } from 'react';
import { useFormContext } from 'react-hook-form';
import { hasDuplicates, sum } from '../../utils/array';
import { VatDetailsFormValues as FormValues } from './interfaces';

const DUPLICATE_ERROR = 'You cannot select the same country more than once';
const MAX_ERROR = 'The Total % of Turnover Cannot Exceed 100%';
const MIN_ERROR = 'The % of turnover must be at least 1%';
const REQUIRED_ERROR = 'Please fill in country of trade and % of turnover';

export function useTurnoverErrorMessage() {
  const { errors } = useFormContext<FormValues>();
  const validErrors = errors.markets?.filter(Boolean) || [];

  const isMissing = validErrors.find(
    _ => _.country?.type === 'required' || _.percentage?.type === 'required',
  );

  if (isMissing) {
    return REQUIRED_ERROR;
  }

  const isTooLow = validErrors.find(_ => _.percentage?.type === 'min');
  if (isTooLow) {
    return MIN_ERROR;
  }

  const isTooHigh = validErrors.find(_ => _.percentage?.type === 'max');
  if (isTooHigh) {
    return MAX_ERROR;
  }

  return errors.isValid?.message?.toString();
}

export function useCustomValidation() {
  const { triggerValidation, register, watch } = useFormContext<FormValues>();

  const markets = watch('markets');
  const validateMarkets = useCallback(() => {
    const countries = markets.map(_ => _.country).filter(Boolean);
    if (hasDuplicates(countries)) {
      return DUPLICATE_ERROR;
    }

    const total = sum(markets, _ => parseInt(_.percentage, 10));
    return total > 100 ? MAX_ERROR : undefined;
  }, [markets]);

  // Register the "virtual" field that holds the validation state
  useEffect(() => {
    register('isValid', { validate: validateMarkets });
  }, [register, validateMarkets]);

  // Validate each time the markets change
  useEffect(() => {
    triggerValidation('isValid');
  }, [markets, triggerValidation]);
}
