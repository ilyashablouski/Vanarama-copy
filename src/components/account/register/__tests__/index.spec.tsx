import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Register from '../index';
import { findForm, fillInputField } from '../../../../services/utils/test-helpers';

const email = "email";
const password = "password";

describe('<Register />', () => {
  const register = jest.fn();

  it('renders correctly with registration', () => {
    const wrapper = mount(<Register register={ register } />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Submit', () => {
    let register;
    let wrapper;

    beforeEach(() => {
      register = jest.fn();
      wrapper = shallow(<Register register={ register } />);
      fillRegisterFields(wrapper);
      findForm(wrapper)
        .simulate('submit', { preventDefault () {} });
    });

    afterEach(() => {
      jest.restoreAllMocks()
    });

    it('it is called once',()=>{
      expect(register).toHaveBeenCalledTimes(1);
    });

    it('it is called with correct parameters',()=>{
      expect(register).toBeCalledWith(email, password)
    });
  });

  describe('HandleChange', () => {
    let register;
    let wrapper;

    beforeEach(() => {
      register = jest.fn();
      wrapper = shallow(<Register register={ register } />);
    });

    afterEach(() => {
      jest.restoreAllMocks()
    });

    it('it has correct state [email]',()=> {
      fillRegisterFields(wrapper);
      expect(wrapper.state(email)).toEqual(email);
    });

    it('it has correct state [password]',()=> {
      fillRegisterFields(wrapper);
      expect(wrapper.state(password)).toEqual(password);
    });

    it('it has incorrect state [email]',()=> {
      expect(wrapper.state('email')).toEqual('');
    });

    it('it has incorrect state [password]',()=> {
      expect(wrapper.state('password')).toEqual('');
    });
  });
});

function fillRegisterFields(wrapper: any) {
    fillInputField(wrapper, 'registerEmail', { name: email, value: email })
    fillInputField(wrapper, 'registerPassword', { name: password, value: password })
}
