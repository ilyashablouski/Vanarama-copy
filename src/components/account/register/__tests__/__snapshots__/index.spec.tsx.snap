// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders correctly with registration 1`] = `
<Register
  register={[MockFunction]}
>
  <section>
    <form
      className="form"
      id="loginForm"
      onSubmit={[Function]}
    >
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="registerEmail"
                    label="Your Email"
                    name="email"
                    type="text"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="registerEmail-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="registerEmail"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Email
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="registerEmail"
                          name="email"
                          onChange={[Function]}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="registerPassword"
                    label="Your Password"
                    name="password"
                    type="password"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="registerPassword-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="registerPassword"
                      >
                        <span
                          className="Text -primary"
                        >
                          Your Password
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="registerPassword"
                          name="password"
                          onChange={[Function]}
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    handleChange={[Function]}
                    id="registerRepeatPassword"
                    label="Repeat Password"
                    name="repeatPassword"
                    type="password"
                    value=""
                  >
                    <div
                      className="TextInput__wrapper -Form__item"
                      id="registerRepeatPassword-wrapper"
                    >
                      <label
                        className="TextInput__label"
                        htmlFor="registerRepeatPassword"
                      >
                        <span
                          className="Text -primary"
                        >
                          Repeat Password
                        </span>
                      </label>
                      <div
                        className="TextInput__field-wrapper"
                      >
                        <input
                          className="TextInput -regular"
                          id="registerRepeatPassword"
                          name="repeatPassword"
                          onChange={[Function]}
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
      <Row
        align="normal"
        component="div"
        debug={false}
        gutterWidth={null}
        justify="start"
        nogutter={false}
        nowrap={false}
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "normal",
              "display": "flex",
              "flexGrow": 0,
              "flexShrink": 0,
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": "16px",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <Col
            component="div"
            debug={false}
            lg={null}
            md={null}
            offset={Object {}}
            pull={Object {}}
            push={Object {}}
            sm={null}
            style={Object {}}
            width={null}
            xl={null}
            xs={null}
          >
            <ScreenClassResolver>
              <ScreenClassProvider
                fallbackScreenClass={null}
                useOwnWidth={false}
              >
                <div
                  style={
                    Object {
                      "boxSizing": "border-box",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 0,
                      "left": "auto",
                      "marginLeft": "0%",
                      "maxWidth": "100%",
                      "minHeight": "1px",
                      "paddingLeft": "15px",
                      "paddingRight": "15px",
                      "position": "relative",
                      "right": "auto",
                      "width": Object {
                        "lg": null,
                        "md": null,
                        "sm": null,
                        "xl": null,
                        "xs": null,
                      },
                    }
                  }
                >
                  <Component
                    color="primary"
                    fill="solid"
                    id="registerButton"
                    label="Register"
                    size="regular"
                    tabIndex={0}
                    type="submit"
                  >
                    <button
                      className="Button -primary -regular -solid"
                      id="registerButton"
                      tabIndex={0}
                      type="submit"
                    >
                      <div
                        className="Button__inner"
                      >
                        Register
                      </div>
                    </button>
                  </Component>
                </div>
              </ScreenClassProvider>
            </ScreenClassResolver>
          </Col>
        </div>
      </Row>
    </form>
  </section>
</Register>
`;
